hooks/tk-multi-reviewsubmission/preprocess_nuke.py

```diff
@@ -18,6 +18,21 @@ def get_processed_script(self, nuke_script_path, **kwargs):
             # nothing to replace, nothing to do here
             return nuke_script_path
 
+        # TODO: publisher.util.get_publish_name() does this much better
+        replace_data["file_base_name"] = os.path.basename(replace_data["path"]).split('.')[0]
+
+        if self.context.entity:
+            sg_entity_type = self.context.entity["type"]
+            sg_filters = [["id", "is", self.context.entity["id"]]]
+
+            # TODO: should we get this list of fields from an app setting? In case we need different fields per show.
+            sg_fields = ["smart_cut_summary_display",
+                         "sg_client_name",
+                         "sg_lens___primary"]
+            replace_data.update(self.shotgun.find_one(sg_entity_type,
+                                                      filters=sg_filters,
+                                                      fields=sg_fields))
+
```

Comment:
Moved this here from app.py because this data is only used for token replacement. This does mean we are making an additional find call, but this does seem to be the right place to get this data.